### DISCUSSION TITLE
Node support for apt

### DISCUSSION THREAD

**Discussion by happybanana9** [nodejs/help#4595] [Q&A] [answered]
Hi, I wanted to download node for my ubuntu server, but currently only external package managers (nvm, fnm) and prebuilt binaries are supported.
Would it be possible to make a .deb package for debian based distros for download?
From that you could just use apt install ./node.deb. No external package managers.

**Comment by thinkingdev9** [comment, marked as answer]
Hi, @happybanana9 . Merry Christmas!
You have to follow steps below to fix this issue.

```
sudo apt update
sudo apt install -y build-essential fakeroot dpkg-dev
```

Download Node.js Source:
Get the source code for the version of Node.js you want to package.

```
VERSION=v16.16.0  # Replace with your desired version
wget https://nodejs.org/dist/$VERSION/node-$VERSION.tar.gz
tar -xzf node-$VERSION.tar.gz
cd node-$VERSION
```

Build Node.js from source.

```
./configure
make -j$(nproc)
```

Install Node.js into a directory which will be used to create the .deb package.

```
mkdir -p ../nodejs-staging/usr/local
make install DESTDIR=$(pwd)/../nodejs-staging/usr/local
```

Create the necessary files for the .deb package.

```
    mkdir -p ../nodejs-staging/DEBIAN

cat <<EOF > ../nodejs-staging/DEBIAN/control
Package: nodejs
Version: ${VERSION:1}
Section: web
Priority: optional
Architecture: amd64
Depends: libc6 (>= 2.17), libstdc++6 (>= 5.2)
Maintainer: Your Name youremail@example.com
Description: Node.js - JavaScript runtime built on Chrome's V8 JavaScript engine
Node.js is a platform built on Chrome's JavaScript runtime for easily building fast, scalable network applications.
EOF
```

Build the .deb Package:

```
cd ../
dpkg-deb --build nodejs-staging
```

Now you can install the .deb package using apt.

```
    sudo apt install ./nodejs-staging.deb
```
